test/extensions/io_socket/user_space/io_socket_handle_impl_platform_test.cc

```diff
@@ -0,0 +1,63 @@
+#include "envoy/common/platform.h"
+#include "envoy/event/file_event.h"
+
+#include "extensions/io_socket/user_space/io_socket_handle_impl.h"
+
+#include "test/mocks/event/mocks.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace IoSocket {
+namespace UserSpace {
+namespace {
+
+using testing::NiceMock;
+
+class MockFileEventCallback {
+public:
+  MOCK_METHOD(void, called, (uint32_t arg));
+};
+
+// Explicitly mark the test failing on windows and will be fixed.
+class BufferedIoSocketHandlePlatformTest : public testing::Test {
+public:
+  BufferedIoSocketHandlePlatformTest() {
+    first_io_handle_ = std::make_unique<IoSocketHandleImpl>();
+    second_io_handle_ = std::make_unique<IoSocketHandleImpl>();
+    first_io_handle_->setPeerHandle(second_io_handle_.get());
+    second_io_handle_->setPeerHandle(first_io_handle_.get());
+  }
+
+  ~BufferedIoSocketHandlePlatformTest() override {
+    if (first_io_handle_->isOpen()) {
+      first_io_handle_->close();
+    }
+    if (second_io_handle_->isOpen()) {
+      second_io_handle_->close();
+    }
+  }
+
+  std::unique_ptr<IoSocketHandleImpl> first_io_handle_;
+  std::unique_ptr<IoSocketHandleImpl> second_io_handle_;
+  NiceMock<Event::MockDispatcher> dispatcher_;
+  MockFileEventCallback cb_;
+};
+
+TEST_F(BufferedIoSocketHandlePlatformTest, CreatePlatformDefaultTriggerTypeFailOnWindows) {
```

Comment:
What about the test below fails on windows?  I expect PlatformDefaultTriggerType to be emulated edge, which does work on Windows.

Reply:
Please enable this test on windows and see if it runs.  If it doesn't, post more info about how it fails.

I think that the extra readable event scheduled in this case is probably benign.  It's possible that the SSL code should ask the iohandle about what mode its running in.  the one you implemented always runs on Edge mode even on windows.

Reply:
> I think that the extra readable event scheduled in this case is probably benign. It's possible that the SSL code should ask the iohandle about what mode its running in. the one you implemented always runs on Edge mode even on windows.

Nice. Thanks!